src/library/dlgtrackinfomulti.cpp

```diff
@@ -0,0 +1,833 @@
+            &DlgTrackInfoMulti::slotCoverInfoSelected);
+
+    connect(m_pWCoverArtMenu,
+            &WCoverArtMenu::reloadCoverArt,
+            this,
+            &DlgTrackInfoMulti::slotReloadCoverArt);
+}
+
+void DlgTrackInfoMulti::slotApply() {
+    saveTracks();
+}
+
+void DlgTrackInfoMulti::slotOk() {
+    slotApply();
+    clear();
+    accept();
+}
+
+void DlgTrackInfoMulti::slotCancel() {
+    clear();
+    reject();
+}
+
+void DlgTrackInfoMulti::loadTracks(const QList<TrackPointer>& pTracks) {
+    clear();
+
+    if (pTracks.isEmpty()) {
+        return;
+    }
+
+    m_pLoadedTracks.clear();
+    for (const auto& pTrack : pTracks) {
+        m_pLoadedTracks.insert(pTrack.get()->getId(), pTrack);
+    }
+
+    updateFromTracks();
+
+    // Listen to all tracks' changed() signal so we don't need to listen to
+    // individual signals such as cuesUpdates, coverArtUpdated(), etc.
+    connectTracksChanged();
+}
+
+void DlgTrackInfoMulti::updateFromTracks() {
+    const QSignalBlocker signalBlocker(this);
+
+    QList<mixxx::TrackRecord> trackRecords;
+    trackRecords.reserve(m_pLoadedTracks.size());
+    for (auto recIt = m_pLoadedTracks.constBegin();
+            recIt != m_pLoadedTracks.constEnd();
+            recIt++) {
+        const auto pTrack = recIt.value();
+        const auto rec = pTrack.get()->getRecord();
+        trackRecords.append(rec);
+    }
+    replaceTrackRecords(trackRecords);
+
+    // Collect star ratings and track colors
+    // If track value differs from the current value, add it to the list.
+    // If new and current are identical, keep only one.
+    int commonRating = m_trackRecords.first().getRating();
+    for (auto& rec : m_trackRecords) {
+        if (commonRating != rec.getRating()) {
+            commonRating = 0;
+            break;
+        }
+    }
+    // Update the star widget
+    // Block signals to not set the 'modified' flag.
+    m_pWStarRating->blockSignals(true);
+    m_pWStarRating->slotSetRating(commonRating);
+    m_starRatingModified = false;
+    m_pWStarRating->blockSignals(false);
+
+    // Same procedure for the track color
+    mixxx::RgbColor::optional_t commonColor = m_trackRecords.first().getColor();
+    bool multipleColors = false;
+    for (auto& rec : m_trackRecords) {
+        if (commonColor != rec.getColor()) {
+            commonColor = mixxx::RgbColor::nullopt();
+            multipleColors = true;
+            break;
+        }
+    }
+    // Paint the color selector and check the respective color picker button.
+    // Paints a rainbow gardient in case of multiple colors.
+    trackColorDialogSetColorStyleButton(commonColor, multipleColors);
+    m_colorChanged = false;
+    m_newColor = mixxx::RgbColor::nullopt();
+
+    // And the track directory
+    QString commonDir = m_pLoadedTracks.constBegin().value()->getFileInfo().canonicalLocationPath();
+    QStringList dirs;
+    dirs.append(commonDir);
+    for (auto dirIt = m_pLoadedTracks.constBegin();
```

Comment:
I think std::as_const accomplishes this?

Reply:
yup, as Daniel mentioned above.